### PR TITLE
file name validation before upload

### PR DESCRIPTION
**Pull request by batikanu**
Sample file name form validation. Validations for yaml and json file names can be also added if they are required.

### CODE REVIEW

File: src/app/frontend/deploy/upload_controller.js [hunk 1]
```diff
@@ -53,4 +59,15 @@ export default class UploadController {
       this.browseFileFunc_();
     }
   }
+
+  /**
+   * Used to deactivate the invalid file name report if the form is not submitted yet.
+   *
+   * @returns {boolean}
+   */
+  isFileNameError() {
```
Comment (bryk) [inline src/app/frontend/deploy/upload_controller.js]: Can you write a test for this?

**Comment by bryk** [comment]
Looks nice. Just a comment.

File: src/app/frontend/deploy/upload_controller.js (hunk 1 above)
Comment (batikanu) [inline src/app/frontend/deploy/upload_controller.js]: I've added some tests for this method

**Comment by bryk** [comment]
This does not work on `gulp serve:prod`. Please fix :)

File: src/app/frontend/deploy/upload_controller.js [hunk 2]
```diff
@@ -53,4 +59,15 @@ export default class UploadController {
       this.browseFileFunc_();
     }
   }
+
+  /**
+   * Used to deactivate the invalid file name report if the form is not submitted yet.
+   *
+   * @returns {boolean}
```
Comment (bryk) [inline src/app/frontend/deploy/upload_controller.js]: This needs @export :)

Comment (batikanu) [inline src/app/frontend/deploy/upload_controller.js]: done :)

**Comment by bryk** [comment]
Thank you!

**Pull request merged**